packages/taquito/test/contract/rpc-contract-provider.spec.ts

```diff
@@ -1627,4 +1636,233 @@ describe('RpcContractProvider test', () => {
       done();
     });
   });
+
+  describe('stake', () => {
+    it('should produce a reveal and stake operation', async (done) => {
+      const estimate = new Estimate(1240000, 94, 145, 249);
+      mockEstimate.stake.mockResolvedValue(estimate);
+      const result = await rpcContractProvider.stake({
+        amount: 100,
+      });
+      expect(result.raw).toEqual({
+        counter: 0,
+        opOb: {
+          branch: 'test',
+          contents: [
+            revealOp('tz1gvF4cD2dDtqitL3ZTraggSR1Mju2BKFEM'),
+            {
+              amount: '100000000',
+              counter: '2',
+              destination: 'tz1gvF4cD2dDtqitL3ZTraggSR1Mju2BKFEM',
+              fee: '479',
+              gas_limit: '1340',
+              kind: 'transaction',
+              parameters: {
+                entrypoint: 'stake',
+                value: {
+                  prim: 'Unit',
+                },
+              },
+              source: 'tz1gvF4cD2dDtqitL3ZTraggSR1Mju2BKFEM',
+              storage_limit: '94',
+            },
+          ],
+          protocol: 'test_proto',
+          signature: 'test_sig',
+        },
+        opbytes: 'test',
+      });
+      done();
+    });
+
+    it('should throw InvalidAmountError when passed a negative amount', async (done) => {
+      const estimate = new Estimate(1240000, 94, 145, 249);
+      mockEstimate.stake.mockResolvedValue(estimate);
+      expect(async () => {
+        await rpcContractProvider.stake({
+          amount: -100,
+        });
+      }).rejects.toThrowError(InvalidAmountError);
+      done();
+    });
+
+    it('should throw InvalidAmountError when passed zero for amount', async (done) => {
```

Comment:
think this case will contradict with `packages/taquito/src/prepare/prepare-provider.ts:235`, if undefined will be default 0 (seem fine to return rpc error `"proto.018-Proxford.operations.invalid_nonzero_transaction_amount"
` to user )